benchmarks/strings.py

```diff
@@ -1,166 +1,119 @@
-
-    timings[name] = {'vaex': t_vaex, 'pandas': t_pandas}
-
-    if use_dask:
-        t_dask = min(mytimeit(expr+".compute()", N, scope={'df': df_dask}))/(10**args.npandas)
-        print("\tdask", t_dask)
-        timings[name]['dask'] = t_dask
-
-        with dask.config.set(scheduler='processes'):
-            t_dask = min(mytimeit(expr+".compute()", N, scope={'df': df_dask}))/(10**args.npandas)
-            print("\tdask_process", t_dask)
-            timings[name]['dask_process'] = t_dask
-
-if __name__ == '__main__':
-
-    test('capitalize', 'df.s.str.capitalize()')
-    test('cat', 'df.s.str.cat(df.s)')
-    test('contains', 'df.s.str.contains("9", regex=False)')
-    test('contains(regex)', 'df.s.str.contains("9", regex=True)')
-    test('count', 'df.s.str.count("9")')
-    test('endswith', 'df.s.str.endswith("9")')
-    test('find', 'df.s.str.find("4")')
-    test('get', 'df.s.str.get(1)')
-    # test('index', 'df.s.str.index(1, 3)') TODO?
-    test('split+join', 'df.s.str.split(".").str.join("-")')
-    test('len', 'df.s.str.len()')
-    test('ljust', 'df.s.str.ljust(10)')
-    test('lower', 'df.s.str.lower()')
-    test('lstrip', 'df.s.str.lstrip("9")')
-    test('match', 'df.s.str.match("1.*")')
-    test('pad', 'df.s.str.pad(10)')
-    test('repeat', 'df.s.str.repeat(2)')
-    test('replace(default)', 'df.s.str.replace("123", "321")')
-    test('replace(no regex)', 'df.s.str.replace("123", "321", regex=False)')
-    test('replace(regex)', 'df.s.str.replace("1?[45]4", "1004", regex=True)')
-    test('rfind', 'df.s.str.rfind("4")')
-    test('rjust', 'df.s.str.rjust(10)')
-    test('rstrip', 'df.s.str.rstrip("9")')
-    test('slice', 'df.s.str.slice(1, 3)')
-    test('split', 'df.s.str.split(".")')
-    test('startswith', 'df.s.str.startswith("9")')
-    test('strip', 'df.s.str.strip("0")') # issues?
-    test('title', 'df.s.str.title()')
-    test('upper', 'df.s.str.upper()')
-    test('zfill', 'df.s.str.zfill(10)')
-
-    for name, values in timings.items():
-        tv = values['vaex']
-        tp = values['pandas']
-        print(name)
-        if use_dask:
-            td = values['dask']
-            print('\t', tv, tp, tp/tv, td/tv, tp/td)
-        else:
-            print('\t', tv, tp, tp/tv)
-   
-    previous_timings = {}
-    fn = "timings.json"
-    if os.path.exists(fn):
-        with open(fn, "r") as f:
-            previous_timings = json.load(f)
-
-    previous_timings.update(timings)
-    timings = previous_timings
-    print(timings)
-    fn = "timings.json"
-    with open(fn, "w") as f:
-        json.dump(timings, f)
-    print('write', fn)
\ No newline at end of file
+
+import numpy as np
+import vaex
+
+
+class StringsSuite:
+    pretty_name = "Performance of string operations in DataFrames"
+    version = "1"
+    timeout = 3600  # in seconds
+
+    param_names = [
+        'n',  # log number of rows to use
+        'partitions',  # number of partitions to split
+    ]
+    params = ([6], [24])
```

Comment:
@maartenbreddels here I used parameters to showcase how they can be used, and to check if you want to run these benchmarks with different number of filtered rows or partitions. If not - then we can get rid of these parameters, as with single value in each of them it's an overkill.

Reply:
What I am experimenting with now is sth like this:
```
class Binning:
    params = ([10**k for k in [8, 9]], [np.float64])
    param_names = ['N', 'dtype']
    def setup(self, N, dtype):
        basedir = os.environ['VAEX_BENCHMARK_DIR']
        self.df = vaex.open(os.path.join(basedir, 'numerical.hdf5'))[:N]
```

so I generate 1 big file (maybe a setup script that runs outside of the benchmark suite), and then slice what I need from it. But I am not sure I like the VAEX_BENCHMARK_DIR env var. I'd rather make a symlink from benchmarks/data to a data directory specific to the machine. But the CWD doesn't seem to be the benchmarks directory.

Reply:
👍 As I mentioned in another comment I don't see a problem with `~/.vaex/data` as used in `datasets.py`.

Reply:
I missed that, I like that idea!